File: smartlist/src/main/java/com/smartlist/module/user/UserResponseCodes.java
package com.smartlist.module.user;

import com.smartlist.utils.AppResponseCodes;
import org.springframework.http.HttpStatus;
import org.springframework.web.server.ResponseStatusException;

public class UserResponseCodes extends AppResponseCodes {

    private UserResponseCodes(){}

    public static final String USER_EMAIL_ID_EXISTS_DESCRIPTION = "user with email already exists";
    public static final String USER_EMAIL_ALREADY_VERIFIED_DESCRIPTION = "user email already verified";
    public static final String TOKEN_EXPIRED_DESCRIPTION = "token expired";
    public static final String INVALID_TOKEN_DESCRIPTION = "token invalid";
    public static final String INVALID_USER_ID_DESCRIPTION = "invalid user id";
    public static final String INVALID_PASSWORD_DESCRIPTION = "invalid user password";
    public static final String INVALID_USER_EMAIL_ID_DESCRIPTION = "invalid user email id";
    public static final String INVALID_ORGANIZATION_ID_DESCRIPTION = "invalid organization id";
    public static final String INVALID_PROFILE_PHOTO_FILE_DESCRIPTION = "invalid profile photo file id";


    public static ResponseStatusException USER_EMAIL_ID_EXISTS = new ResponseStatusException(HttpStatus.BAD_REQUEST, USER_EMAIL_ID_EXISTS_DESCRIPTION);

    public static ResponseStatusException USER_EMAIL_ALREADY_VERIFIED = new ResponseStatusException(HttpStatus.BAD_REQUEST, USER_EMAIL_ALREADY_VERIFIED_DESCRIPTION);

    public static ResponseStatusException TOKEN_EXPIRED = new ResponseStatusException(HttpStatus.BAD_REQUEST, TOKEN_EXPIRED_DESCRIPTION);

    public static ResponseStatusException INVALID_TOKEN = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_TOKEN_DESCRIPTION);

    public static ResponseStatusException INVALID_USER_ID = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_USER_ID_DESCRIPTION);

    public static ResponseStatusException INVALID_PASSWORD = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_PASSWORD_DESCRIPTION);

    public static ResponseStatusException INVALID_USER_EMAIL_ID = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_USER_EMAIL_ID_DESCRIPTION);

    public static ResponseStatusException INVALID_ORGANIZATION_ID = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_ORGANIZATION_ID_DESCRIPTION);

    public static ResponseStatusException INVALID_PROFILE_PHOTO_FILE_ID = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_PROFILE_PHOTO_FILE_DESCRIPTION);

}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/user/user.response.codes.ts
import { NotFoundException } from '@nestjs/common';
import { BadRequestException } from '@nestjs/common';
import { AppResponseCodes } from 'src/app.response.codes';

export class UserResponseCodes extends AppResponseCodes {
  public static USER_EMAIL_ID_EXISTS: any = new BadRequestException(
    'user with email already exists',
  );

  public static USER_EMAIL_ALREADY_VERIFIED: any = new BadRequestException(
    'user email already verified',
  );

  public static TOKEN_EXPIRED: any = new BadRequestException('token expired');

  public static INVALID_TOKEN: any = new NotFoundException('token invalid');

  public static INVALID_USER_ID: any = new NotFoundException('invalid user id');

  public static INVALID_USER_ID_OR_PASSWORD: any = new NotFoundException(
    'invalid user id or password',
  );

  public static INVALID_USER_EMAIL_ID: any = new NotFoundException(
    'invalid user email id',
  );

  public static INVALID_ORGANIZATION_ID: any = new NotFoundException(
    'invalid organization id',
  );

  public static INVALID__PROFILE_PHOTO_FILE_ID: any = new NotFoundException(
    'invalid profile photo file id',
  );
}
